cmd/juju/secretbackends/add.go

```diff
@@ -152,28 +160,36 @@ func (c *addSecretBackendCommand) readFile(ctx *cmd.Context) (map[string]interfa
 	return attrs, nil
 }
 
-// Run implements cmd.Run.
-func (c *addSecretBackendCommand) Run(ctxt *cmd.Context) error {
-	attrs, err := c.readFile(ctxt)
-	if err != nil {
-		return errors.Trace(err)
-	}
-	for k, v := range c.KeyValueAttrs {
-		attrs[k] = v
-	}
-
+func parseTokenRotate(attrs map[string]interface{}, zeroAllowed bool) (*time.Duration, error) {
 	const tokenRotate = "token-rotate"
 	var tokenRotateInterval *time.Duration
 	tokenRotateStr, ok := attrs[tokenRotate]
 	if ok {
 		delete(attrs, tokenRotate)
 		rotateInterval, err := time.ParseDuration(fmt.Sprintf("%s", tokenRotateStr))
-		if err != nil || (rotateInterval/time.Second) < 60 {
-			return errors.NotValidf("token rotate interval %q", tokenRotateStr)
+		intervalSecs := rotateInterval / time.Second
+		if err != nil || intervalSecs < 60 && (!zeroAllowed || intervalSecs != 0) {
```

Comment:
these should be seperate error messages (one for less than 60s, one for zero) and actually expose the time.ParseDuration error.